lib/modulesync.rb

```diff
@@ -114,7 +121,31 @@ def self.manage_module(puppet_module, module_files, module_options, defaults, op
     if options[:noop]
       Git.update_noop(module_name, options)
     elsif !options[:offline]
-      Git.update(module_name, files_to_manage, options)
+      # Git.update() returns a boolean: true if files were pushed, false if not.
+      pushed = Git.update(module_name, files_to_manage, options)
+      return nil unless pushed
+
+      if options[:pr] && GITHUB_TOKEN
+        # We only do GitHub PR work if the GITHUB_TOKEN variable is set in the environment.
+        repo_path = "#{namespace}/#{module_name}"
+        puts "Submitting PR '#{options[:pr_title]}' on GitHub to #{repo_path} - merges #{options[:branch]} into master"
+        github = Octokit::Client.new(:access_token => GITHUB_TOKEN)
```

Comment:
Would it be wise to cache this object somewhere? Would that allow reusing the network connection or doesn't the gem use that anyway.

Reply:
It looks like the underlying library, Faraday, supports persistent HTTP connections, but it's not entirely clear how to make Octokit use it. This feels like a premature optimization given we're already living with Git connections being torn down each module / shell-out operations running in serial. In my testing I've ran this against low dozens of our repos in a single run and it doesn't feel slower than the existing runs.

Reply:
That's fair. Thanks for considering and let's keep this simple for now.